pkg/api/api.go

```diff
@@ -41,18 +78,65 @@ func CheckDetails(c echo.Context) error {
 	}
 
 	start := time.Now()
-	results, err := cache.PostgresCache.QueryStatus(*q)
+	statuses, err := cache.PostgresCache.QueryStatus(*q)
 	if err != nil {
 		return errorResonse(c, err, http.StatusInternalServerError)
 	}
-	apiResponse := &DetailResponse{
-		RunnerName: runner.RunnerName,
-		Status:     results,
+
+	apiResponse := &CheckRunDetails{
 		Duration:   int(time.Since(start).Milliseconds()),
+		Latency:    measureLatency(statuses),
+		RunnerName: runner.RunnerName,
+		Status:     aggregateWindow(statuses, 100),
+		// TODO: uptime
 	}
+
 	return c.JSON(http.StatusOK, apiResponse)
 }
 
+func merge(ts []pkg.Timeseries) pkg.Timeseries {
+	if len(ts) == 0 {
+		return pkg.Timeseries{}
+	}
+
+	output := ts[0] // TODO: should use all the data of the first item
+	var sum int
+	for _, t := range ts {
+		sum += t.Duration
+	}
+
+	output.Duration = int(math.Ceil(float64(sum) / float64(len(ts))))
+	return output
+}
+
+func aggregateWindow(ts []pkg.Timeseries, maxDataPoints int) []pkg.Timeseries {
+	if len(ts) <= maxDataPoints {
```

Comment:
This needs to be at the db layer.

Reply:
By that do you mean in the `cache` package or in the SQL itself?